content/docs/1.10.0/advanced-resources/rebuilding/_index.md

```diff
@@ -1,4 +1,188 @@
 ---
 title: Replica Rebuilding
 weight: 6
----
\ No newline at end of file
+---
+- [Replica Rebuilding Workflow](#replica-rebuilding-workflow)
+  - [Full Replica Rebuilding](#full-replica-rebuilding)
+  - [Delta Replica Rebuilding](#delta-replica-rebuilding)
+  - [Fast Replica Rebuilding](#fast-replica-rebuilding)
+- [Factors That Affect Rebuilding Performance](#factors-that-affect-rebuilding-performance)
+- [Use Cases](#use-cases)
+  - [Node Reboot During Upgrade](#node-reboot-during-upgrade)
+  - [Short-Term Node Drain](#short-term-node-drain)
+- [Relevant Settings](#relevant-settings)
+  - [Settings Trade-Off Analysis](#settings-trade-off-analysis)
+
+When Longhorn detects a failed or deleted replica, it automatically initiates a rebuilding process. This document outlines the replica rebuilding workflow, including **full**, **delta**, and **fast** rebuilding methods. It also explains the limitations associated with each method.
+
+**Rebuilding will not start in the following scenarios**:
+
+- The volume is migrating to another node.
+- The volume is an old restore/DR volume.
+- The volume is expanding in size.
+
+## Replica Rebuilding Workflow
+
+Replica rebuilding may be triggered in the following scenarios:
+
+- A node is rebooted or evicted.
+- A replica becomes unhealthy or is deleted.
+
+ {{< figure alt="Replica Rebuilding Flow Diagram" src="/img/diagrams/architecture/replica-rebuilding-flow.png" >}}
+
+1. Mark the target replica with `WO` (Write-Only) mode.
+2. Create a new snapshot to serve as the volume head reference point for data integrity checks.
+3. Generate the synchronization file list for the volume head and snapshot files.
+4. Launch a receiver server on the target replica for each snapshot, then instruct the source replica to begin data synchronization.
+5. For each snapshot synchronization:
+    - Check whether the snapshot file exists in the target replica's data directory:
+        - If **NO**, transfer the entire snapshot data from the source replica to the target replica. See [Full Replica Rebuilding](#full-replica-rebuilding)
+        - If **YES**, check whether the snapshot checksum files exist and are identical between the target and source replicas:
+            - If **YES**, Longhorn skips transferring that snapshot’s data. This optimization reduces CPU usage, disk I/O, network I/O, and overall rebuild time. See [Fast Replica Rebuilding](#fast-replica-rebuilding)
+            - If **NO**, Longhorn calculates and compares block-by-block checksums for the snapshot file using the SHA-512 algorithm. If mismatches are found, only the differing data blocks are synchronized. See [Delta Replica Rebuilding](#delta-replica-rebuilding)
```

Comment:
Same. Do we need to mention the workflow differences for the v2 data engine?

Reply:
Added for `v1 data engine` in item 5 title. Thanks